common/deftype/defvalue.cpp

```diff
@@ -435,8 +435,10 @@ void MemoryValue::pushDecimalValue()
 
 void MemoryValue::serialize(MemoryBuffer &tgt)
 {
-    tgt.append(val.length());
-    tgt.append(val.length(), val.get());
+    size32_t serialLen = (size32_t)val.length();
```

Comment:
MemoryValue::deserialize reads the size32_t back then calls checked_malloc(size_t, ..

Is the use of size32_t here because MemoryValue wants to be limited to 32bit and conserve serialization space?
If not, would look cleaner if just serialized + deserialized size_t
Or is this a backward compatibility thing?